PubNub/Data/Service Objects/PNStatus.m

```diff
@@ -152,14 +152,14 @@ - (instancetype)initForOperation:(PNOperationType)operation
     if ((self = [super initForOperation:operation completedWithTask:task processedData:processedData
                         processingError:error])) {
         
-        _error = (error != nil || self.statusCode != 200);
+        _error = (error != nil || self.statusCode >= 400);
         
         if (_error && ![self.serviceData count]) {
             [self updateData:[self dataFromError:error]];
         }
         
         // Check whether status should represent acknowledgment or not.
-        if (self.statusCode == 200 && !_error) {
+        if (self.statusCode < 400 && !_error) {
```

Comment:
Does (should) it matter that redirection status codes (3xx) are acknowledged?

Reply:
@davidnub With NSULRSession I don't get any redirects, because they not final resource which I requested. This status code is result of request processing, not intermediate state (it will follow redirects on it's own).